targets/XPlatCppSdk/source/code/source/playfab/PlayFabAndroidHttpPlugin.cpp

```diff
@@ -0,0 +1,743 @@
+#include <stdafx.h>
+
+#include <jni.h>
+#include <android/log.h>
+
+#include <playfab/PlayFabAndroidHttpPlugin.h>
+#include <playfab/PlayFabSettings.h>
+#include <stdexcept>
+#include <vector>
+
+namespace PlayFab
+{
+    struct PlayFabAndroidHelper
+    {
+    public:
+        PlayFabAndroidHelper() :
+                jvm(nullptr),
+                httpRequestClass(nullptr)
+        {
+
+        }
+
+        virtual ~PlayFabAndroidHelper()
+        {
+            if (httpRequestClass)
+            {
+                JNIEnv* jniEnv;
+                if(GetJniEnv(&jniEnv))
+                {
+                    jniEnv->DeleteGlobalRef(httpRequestClass);
+                }
+                httpRequestClass = nullptr;
+            }
+            this->jvm = nullptr;
+        }
+
+        bool Initialize(JavaVM* jvm)
+        {
+            assert(jvm);
+            this->jvm = jvm;
+
+            JNIEnv* jniEnv;
+            this->jvm->GetEnv(reinterpret_cast<void**>(&jniEnv), JNI_VERSION_1_6);
+            assert(jniEnv);
+            if (jniEnv == nullptr)
+            {
+                return false;
+            }
+
+            jclass localClass = jniEnv->FindClass("com/microsoft/xplatcppsdk/HttpRequest");
+            assert(localClass);
```

Comment:
>assert(localClass); [](start = 12, length = 19)

are these debug asserts or retail asserts ?

Reply:
This is debug assert if there defined NDEBUG. But I didn't implemented Release mode, so this will make assert on release mode.

---
In reply to: [273698342](https://github.com/PlayFab/SDKGenerator/pull/470#discussion_r273698342) [](ancestors = 273698342)